trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,228 @@
+    found the priors below to be highly effective for objective functions defined over the unit
+    hypercube. For objective functions with different scaling, other priors will likely be more
+    appropriate. Using priors allows for using maximum a posteriori estimate of these kernel
+    parameters during model fitting.
+
+    :param data: Dataset from the initial design, used for estimating the variance of observations.
+    :param noise_free: If set to `True` Gaussian likelihood paramater is set to
+        non-trainable. By default set to `False`.
+    :param priors: If set to `True` (default) priors are set for kernel parameters.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
+    else:
+        noise_variance = likelihood_variance
+
+    model = gpflow.models.GPR(data.astuple(), kernel, mean, noise_variance)
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+
+    return model
+
+
+def build_sgpr(
+    data: Dataset,
+    search_space: Box,
+    kernel_priors: bool = True,
+    noise_free: bool = False,
+    likelihood_variance: Optional[float] = None,
+    num_inducing_points: Optional[int] = None,
+    trainable_inducing_points: bool = False,
+) -> SGPR:
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    inducing_points = _get_inducing_points(search_space, num_inducing_points)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
+    else:
+        noise_variance = likelihood_variance
+
+    model = gpflow.models.SGPR(
+        data.astuple(), kernel, inducing_points, mean_function=mean, noise_variance=noise_variance
+    )
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+    gpflow.set_trainable(model.inducing_variable, trainable_inducing_points)
+
+    return model
+
+
+def build_vgp(
+    data: Dataset,
+    search_space: Box,
+    classification: bool = False,
+    kernel_priors: bool = True,
+    noise_free: bool = False,
+    likelihood_variance: Optional[float] = None,
+) -> VGP:
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    if classification:
+        if noise_free:
+            empirical_variance = 100.0
+        else:
+            empirical_variance = 1.0
+        model_likelihood = gpflow.likelihoods.Bernoulli()
+    else:
+        if likelihood_variance is None:
+            noise_variance = 0.1 * empirical_variance
+        else:
+            noise_variance = likelihood_variance
+        model_likelihood = gpflow.likelihoods.Gaussian(noise_variance)
```

Comment:
Do we use this model anywhere? I really don't see the point of a VGP with Gaussian likelihood... it's just a poorly defined GPR.

Reply:
true, we use it only for classification - @henrymoss do you also agree we don't need VGP for regression?